### PR TITLE
changelist implementation

### PR DESCRIPTION
**Pull request by endophage**
This is just the changelist implementation with no integration into the client.

Tests on the way but wanted to get this PR started.

### CODE REVIEW

File: client/changelist/change.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package changelist
+
+import (
+	"encoding/json"
+
+	"github.com/endophage/gotuf/data"
+)
+
+type tufChange struct {
+	Actn       string `json:"action"`
```
Comment (NathanMcCauley) [inline client/changelist/change.go]: No need to abbreviate

Comment (endophage) [inline client/changelist/change.go]: Can't have a method and field with the same name :-(

Comment (NathanMcCauley) [inline client/changelist/change.go]: ok, i'm wrong :)

File: client/changelist/changelist.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	}
+	cl.file.Write([]byte("\n"))
+	cl.file.Sync()
+	return nil
+}
+
+// Clear empties the changelist file. It does not currently
+// support archiving
+func (cl *fileChangelist) Clear(archive string) error {
+	cl.file.Seek(0, 0)  // seek to start
```
Comment (NathanMcCauley) [inline client/changelist/changelist.go]: Maybe add a TODO about archiving?

**Comment by NathanMcCauley** [comment]
LGTM

**Comment by endophage** [comment]
Cheers for the review. Will be resubmitting with integration included later.

File: client/changelist/interface.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+	// Clear empties the current change list.
+	// Archive may be provided as a directory path
+	// to save a copy of the changelist in that location
+	Clear(archive string) error
+}
+
+type Change interface {
+	// "create","update", or "delete"
+	Action() string
```
Comment (endophage) [inline client/changelist/interface.go]: from mccauley: Create an enum for Action

**Pull request closed without merging**